presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java

```diff
@@ -0,0 +1,181 @@
+                    .forEach(builder::add);
+        }
+        return new FixedSplitSource(builder.build());
+    }
+
+    private ExtendedFileSystem getFileSystem(ConnectorSession session, HudiTableHandle table)
+    {
+        HdfsContext hdfsContext = new HdfsContext(
+                session,
+                table.getSchemaName(),
+                table.getTableName(),
+                table.getPath(),
+                false);
+        try {
+            return hdfsEnvironment.getFileSystem(hdfsContext, new Path(table.getPath()));
+        }
+        catch (IOException e) {
+            throw new PrestoException(HUDI_FILESYSTEM_ERROR, "Could not open file system for " + table, e);
+        }
+    }
+
+    private HudiSplit createHudiSplit(HudiTableHandle table, FileSlice slice, HudiPartition partition)
+    {
+        HoodieBaseFile hoodieBaseFile = slice.getBaseFile().get();
+        String baseFilePath = hoodieBaseFile.getPath();
+        return new HudiSplit(
+                table,
+                slice.getBaseInstantTime(),
+                partition,
+                baseFilePath,
+                0,
+                hoodieBaseFile.getFileLen(),
+                slice.getLogFiles().map(l -> l.getPath().toString()).collect(Collectors.toList()),
+                ImmutableList.of(),
+                NodeSelectionStrategy.NO_PREFERENCE);
+    }
+
+    public static HudiPartition getHudiPartition(ExtendedHiveMetastore metastore, MetastoreContext context, HudiTableLayoutHandle tableLayout, String partitionName)
+    {
+        String databaseName = tableLayout.getTable().getSchemaName();
+        String tableName = tableLayout.getTable().getTableName();
+        List<HudiColumnHandle> partitionColumns = tableLayout.getPartitionColumns();
+
+        if (partitionColumns.isEmpty()) {
+            // non-partitioned tableLayout
+            Table table = metastore.getTable(context, databaseName, tableName)
+                    .orElseThrow(() -> new PrestoException(HUDI_INVALID_METADATA, format("Table %s.%s expected but not found", databaseName, tableName)));
+            return new HudiPartition(partitionName, ImmutableList.of(), ImmutableMap.of(), table.getStorage(), tableLayout.getDataColumns());
+        }
+        else {
+            // partitioned tableLayout
+            List<String> partitionValues = extractPartitionValues(partitionName);
+            checkArgument(partitionColumns.size() == partitionValues.size(),
+                    format("Invalid partition name %s for partition columns %s", partitionName, partitionColumns));
+            Partition partition = metastore.getPartition(context, databaseName, tableName, partitionValues)
+                    .orElseThrow(() -> new PrestoException(HUDI_INVALID_METADATA, format("Partition %s expected but not found", partitionName)));
+            Map<String, String> keyValues = zipPartitionKeyValues(partitionColumns, partitionValues);
```

Comment:
Not sure if you're considering different [partition extractors](https://github.com/apache/hudi/blob/ecbad9526ae2c4531c00618f178fd34e10a21db3/hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/PartitionValueExtractor.java) in hudi-hive sync. Depending on partition extractor, partition values can change.

Reply:
We are now retrieving partition metadata through HiveMetaStore, where the partition values stored have been extracted by some partition extractors from their partition locations and thus we dont need to take consisder of the partition extractors.